rust_src/src/frames.rs

```diff
@@ -69,6 +69,15 @@ impl LispFrameRef {
             return 0;
         }
     }
+
+    /// Replaces FRAME_WINDOW_P
+    pub fn is_gui_window(self) -> bool {
+        #[allow(unused_variables)]
+        let method = self.output_method();
+        (cfg!(feature = "window-system-x11") && method == output_method::output_x_window)
```

Comment:
https://github.com/alexcrichton/cfg-if might be a little cleaner. Worth a look.

Reply:
yeah, that definitely looks like the way to go, especially with code that should't be compiled. Here there's not much difference but I've already had an instance of `#[cfg(foo)]` + `#[cfg(not(foo))]`